`TypeError: require is not a function` when running `npm run docs:watch` locally

I've tried to contribute to the documentation, but when I run `npm run docs:watch`, I only get a blank page with the following error:

```
Uncaught TypeError: require is not a function
    at index.js:9:9
```

![image](https://github.com/vuejs/eslint-plugin-vue/assets/31937175/d28628da-a2e3-455b-a094-1b2702f1bc79)

What I did:
- installed dependencies with `npm install`
- ran `npm run docs:watch`

I've also tried to build the library with `npm run tsc`, but got the following error:

```
lib/rules/order-in-components.js:265:25 - error TS7006: Parameter 'fixer' implicitly has an 'any' type.

265     function* handleFix(fixer, propertyNode, unorderedPropertyNode) {
                            ~~~~~


Found 1 error in lib/rules/order-in-components.js:265
```

I'm on:
- Windows 10
- Node v20.15.0

--- a/docs/.vitepress/build-system/build.mts
+++ b/docs/.vitepress/build-system/build.mts
@@ -2,7 +2,7 @@
  * Pre-build cjs packages that cannot be bundled well.
  */
 import esbuild from 'esbuild'
-import path from 'path'
+import path from 'pathe'
 import fs from 'fs'
 import { fileURLToPath } from 'url'
 

--- a/docs/.vitepress/config.mts
+++ b/docs/.vitepress/config.mts
@@ -1,6 +1,6 @@
 import type { DefaultTheme } from 'vitepress'
 import { defineConfig } from 'vitepress'
-import path from 'path'
+import path from 'pathe'
 import { fileURLToPath } from 'url'
 import { viteCommonjs, vitePluginRequireResolve } from './vite-plugin.mjs'
 

--- a/docs/.vitepress/vite-plugin.mts
+++ b/docs/.vitepress/vite-plugin.mts
@@ -1,5 +1,5 @@
 import type { UserConfig } from 'vitepress'
-import path from 'path'
+import path from 'pathe'
 import { fileURLToPath } from 'url'
 import esbuild from 'esbuild'
 type Plugin = Extract<

--- a/lib/rules/order-in-components.js
+++ b/lib/rules/order-in-components.js
@@ -259,6 +259,7 @@ module.exports = {
     }
 
     /**
+     * @param {RuleFixer} fixer
      * @param {Property} propertyNode
      * @param {Property} unorderedPropertyNode
      */

--- a/package.json
+++ b/package.json
@@ -92,6 +92,7 @@
     "markdownlint-cli": "^0.41.0",
     "mocha": "^10.3.0",
     "nyc": "^17.0.0",
+    "pathe": "^1.1.2",
     "prettier": "^3.2.5",
     "typescript": "^5.3.3",
     "vitepress": "^1.0.0-rc.42"

